plugins/statistics.koplugin/main.lua

```diff
@@ -2822,11 +2840,15 @@ function ReaderStatistics:onReadingResumed()
 end
 
 function ReaderStatistics:onReaderReady(config)
-    self.data = config:readSetting("stats", { performance_in_pages = {} })
-    self.doc_md5 = config:readSetting("partial_md5_checksum")
-    -- we have correct page count now, do the actual initialization work
-    self:initData()
-    self.view.footer:maybeUpdateFooter()
+    UIManager:nextTick(function()
+        if self.settings.is_enabled then
+            self.data = config:readSetting("stats", { performance_in_pages = {} })
+            self.doc_md5 = config:readSetting("partial_md5_checksum")
+            -- we have correct page count now, do the actual initialization work
+            self:initData()
+            self.view.footer:maybeUpdateFooter()
+        end
+    end)
```

Comment:
Hard (and no time from me) to check for any side effects, but be ultra sure there's no issue delaying all of this until nextTick (including accounting the opening page for statistics).
There may be lots of stuff happening between ReaderReady and its next tick (ie. all the postReaderReadyCallback stuff).

Reply:
> all the postReaderReadyCallback stuff

3 calls only, and it looks more correct to begin the staistics after them:
https://github.com/koreader/koreader/blob/9f61c230a371ddefa87b9929fcc678289258cfad/frontend/apps/reader/modules/readerrolling.lua#L190-L196
https://github.com/koreader/koreader/blob/9f61c230a371ddefa87b9929fcc678289258cfad/frontend/apps/reader/modules/readerlink.lua#L121-L124
https://github.com/koreader/koreader/blob/9f61c230a371ddefa87b9929fcc678289258cfad/frontend/apps/reader/modules/readerannotation.lua#L134-L139